runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -168,8 +168,12 @@ def update_state_from_location_header(request, polling_state, custom_deserializa
     # Updates polling state from Azure async operation header.
     # @param polling_state [MsRestAzure::PollingState] polling state.
     #
-    def update_state_from_azure_async_operation_header(request, polling_state)
-      result = get_async_with_async_operation_deserialization(request)
+    def update_state_from_azure_async_operation_header(request, polling_state, custom_deserialization_block)
+      if custom_deserialization_block.nil?
+        result = get_async_with_async_operation_deserialization(request)
```

Comment:
It'd be great if you can combing two methods (get_async_with_async_operation_deserialization & get_async_with_custom_deserialization) into one that can handle `custom_deserialization_block` if provided.